### PR TITLE
Random improvements (MJCF, error messages, gravity)

### PR DESCRIPTION
**Pull request by olegklimov**
I will be adding here any additional improvements, and I'll also always rebase on top. @erwincoumans please merge whenever you feel the right moment.

### CODE REVIEW

File: examples/SharedMemory/PhysicsServerCommandProcessor.cpp [hunk 1]
```diff
@@ -868,7 +868,8 @@ struct PhysicsServerCommandProcessorInternalData
 
 
 	btScalar m_physicsDeltaTime;
-    btScalar m_numSimulationSubSteps;
+	btScalar m_numSimulationSubSteps;
+	btVector3 m_gravity;
```
Comment (erwincoumans) [inline examples/SharedMemory/PhysicsServerCommandProcessor.cpp]: This is not desired: resetSimulation should reset the state as startup, undo all things the user did (loading URDF files, setting custom settings such as gravity). It should be trivial for a user to set the gravity after doing a reset.

**Comment by erwincoumans** [comment]
The improvements are OK, except for remembering the gravity vector: resetSimulation should reset the known startup state. Can you please revert that change, and I'll merge the PR. Thanks!

**Comment by olegklimov** [comment]
> remembering the gravity vector

Yeah that was silly of me. I'm seeing things that are not there.

Let me rebase this...

**Comment by olegklimov** [comment]
image
image

Capsule/cylinder length is used in many places, can be a bug somewhere... Works for me, but I haven't check tiny renderer for example.

File: examples/SharedMemory/PhysicsServerCommandProcessor.cpp [hunk 2]
```diff
@@ -3808,6 +3808,12 @@ bool PhysicsServerCommandProcessor::processCommand(const struct SharedMemoryComm
 									{
 										linkIndexB = mblB->m_link;
 										objectIndexB = mblB->m_multiBody->getUserIndex2();
+										if (
```
Comment (erwincoumans) [inline examples/SharedMemory/PhysicsServerCommandProcessor.cpp]: Why is this filterneeded? Say the filter is for object 2 and object 3, then we should also allow object 3 and object 2 (the order doesn't matter). This check is already applied in the existing code.

Comment (olegklimov) [inline examples/SharedMemory/PhysicsServerCommandProcessor.cpp]: link filter, not object filter

Comment (erwincoumans) [inline examples/SharedMemory/PhysicsServerCommandProcessor.cpp]: Thanks, that's OK then. I'll review and test the other changes too.

**Comment by erwincoumans** [comment]
LGTM, thanks for the contribution again!

**Pull request merged**